hibernate-envers/src/main/java/org/hibernate/envers/internal/entities/mappergenerator/CollectionMapperResolver.java

```diff
@@ -0,0 +1,17 @@
+package org.hibernate.envers.internal.entities.mappergenerator;
+
+import org.hibernate.envers.internal.entities.mapper.PropertyMapper;
+import org.hibernate.envers.internal.entities.mapper.relation.CommonCollectionMapperData;
+import org.hibernate.envers.internal.entities.mapper.relation.MiddleComponentData;
+import org.hibernate.mapping.Collection;
+
+/**
+ * Interface for classes responsible for generating appropriate collection mappers for hibernate collection data types
+ *
+ * @author Michal Skowronek
+ */
+public interface CollectionMapperResolver {
+    PropertyMapper resolveCollectionMapper(Collection collection, CommonCollectionMapperData commonCollectionMapperData,
```

Comment:
So what I gather thus far is it seems the prime reason you want to provide your own mapper implementations is so that you can override the newly added `initializeResultEntities` methods on these mappers and perform some post-query step in order to hydrate additional information perhaps or alter entity state?

If so, why must this be done inside the Envers mappers rather than inside your persistence code?  For example, using pseudo code:
````
   List results = getAuditReader().doSomeFancyAuditQuery( ... );
   for ( Object result : results ) {
     // initialize or perform any post-op task needed
   }
````
There may be a way we could handle something like this in Envers, but I'm not convinced that the no-op calls in the `PropertyMapper` implementations is the right approach.